lib/ui/views/app_selector/app_selector_viewmodel.dart

```diff
@@ -97,11 +98,51 @@ class AppSelectorViewModel extends BaseViewModel {
         return showSelectFromStorageDialog(context);
       } else if (!await checkSplitApk(packageName) || isRooted) {
         selectApp(app);
+        showVersionNotSupportedDialog(context);
+        locator<PatcherViewModel>().notifyListeners();
         Navigator.pop(context);
       }
     }
   }
 
+  Future showVersionNotSupportedDialog(
+    BuildContext context,
+  ) async {
+    final recommendedVersion = await _patcherAPI.getSuggestedVersion(
+        locator<PatcherViewModel>().selectedApp!.packageName);
```

Comment:
Could you check whether the installed version matches any of the versions in the compatibility list?